atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingIteratorTest.java

```diff
@@ -17,68 +17,83 @@
 import java.util.Iterator;
 import java.util.List;
-import java.util.function.Consumer;
 import java.util.function.ToLongFunction;
-import one.util.streamex.StreamEx;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
+@RunWith(MockitoJUnitRunner.class)
 public final class TrackingIteratorTest {
     private static final String STRING = "test";
-    private static final Iterator<String> ONE_ELEMENT_ITERATOR = List.of(STRING).iterator();
-    private static final ImmutableList<String> STRINGS = ImmutableList.of(
-            "test4", "test4", "test200", "composite", "", "t", "twentyElementString1", "tt", "twentyElementString2");
 
-    // these have to be anonymous inner classes rather than lambdas in order to spy
-    private static final Consumer<Long> NO_OP = new Consumer<>() {
-        @Override
-        public void accept(Long _unused) {}
-    };
-    private static final ToLongFunction<String> STRING_LENGTH_MEASURER = new ToLongFunction<>() {
-        @Override
-        public long applyAsLong(String value) {
-            return value.length();
-        }
-    };
+    @Mock
+    private BytesReadTracker tracker;
+
+    @Mock
+    private ToLongFunction<String> measurer;
 
     @Test
     public void oneElementTrackingIteratorIsWiredCorrectly() {
-        Consumer<Long> tracker = spy(NO_OP);
-        ToLongFunction<String> measurer = spy(STRING_LENGTH_MEASURER);
-        TrackingIterator<String, Iterator<String>> trackingIterator =
-                new TrackingIterator<>(ONE_ELEMENT_ITERATOR, tracker, measurer);
+        long measuredValue = 1L;
+        when(measurer.applyAsLong(anyString())).thenReturn(measuredValue);
+        Iterator<String> trackingIterator = createTrackingIterator(Iterators.singletonIterator(STRING));
+
+        assertThat(trackingIterator).toIterable().containsExactly(STRING);
 
-        assertThat(trackingIterator).toIterable().containsExactlyElementsOf(List.of(STRING));
         verify(measurer).applyAsLong(STRING);
-        verify(tracker).accept(STRING_LENGTH_MEASURER.applyAsLong(STRING));
+        verify(tracker).record(measuredValue);
         verifyNoMoreInteractions(tracker, measurer);
     }
 
     @Test
-    public void multiElementTrackingIteratorIsWiredCorrectly() {
-        ArrayList<Long> consumed = new ArrayList<>();
-        TrackingIterator<String, Iterator<String>> trackingIterator =
-                new TrackingIterator<>(createStringIterator(), consumed::add, STRING_LENGTH_MEASURER);
+    public void trackingIteratorDelegatesNext() {
+        List<String> strings = ImmutableList.of("", "length1", "length2", "test");
+        Iterator<String> trackingIterator = createTrackingIterator(strings.iterator());
+        assertThat(trackingIterator).toIterable().containsExactlyElementsOf(strings);
+    }
+
+    @Test
+    public void trackingIteratorFeedsTracker() {
```

Comment:
nit: `feedsTrackerInOrder`? Admittedly, your operations for tex stats are currently commutative, so this is a bit strong (an equally valid test would be to separately mock applyAsLong for each string in the array, s.t. only `one` will return `1L` e.t.c). 

Are you considering adding non-commutative stats?